enterprise/security/src/main/java/org/neo4j/server/security/enterprise/auth/plugin/PluginRealm.java

```diff
@@ -148,8 +149,16 @@ protected AuthorizationInfo doGetAuthorizationInfo( PrincipalCollection principa
     {
         if ( authorizationPlugin != null )
         {
-            org.neo4j.server.security.enterprise.auth.plugin.spi.AuthorizationInfo authorizationInfo =
-                    authorizationPlugin.authorize( getPrincipalAndRealmCollection( principals ) );
+            org.neo4j.server.security.enterprise.auth.plugin.spi.AuthorizationInfo authorizationInfo;
+            try
+            {
+                 authorizationInfo = authorizationPlugin.authorize( getPrincipalAndRealmCollection( principals ) );
+            }
+            catch ( AuthorizationExpired e )
+            {
+                throw new AuthExpirationException(
+                        "Plugin '" + getName() + "' authorization info expired: " + e.getMessage() );
```

Comment:
I'm a fan of attaching cause exceptions. Is there a particular reason we do not want to do that here? (To clarify, I mean sending in the caught exception into the constructor of the thrown exception.)